python/lsst/fgcmcal/fgcmLoadReferenceCatalog.py

```diff
@@ -0,0 +1,336 @@
+            refCat = skyCircle.refCat
+
+        # Select on raw (uncorrected) catalog, where the errors should make more sense
+        goodSources = self.referenceSelector.selectSources(refCat)
+        selected = goodSources.selected
+
+        fgcmRefCat = np.zeros(np.sum(selected), dtype=[('ra', 'f8'),
+                                                       ('dec', 'f8'),
+                                                       ('refMag', 'f4', len(filterList)),
+                                                       ('refMagErr', 'f4', len(filterList))])
+        if fgcmRefCat.size == 0:
+            # Return an empty catalog if we don't have any selected sources
+            return fgcmRefCat
+
+        # The ra/dec native Angle format is radians
+        # We determine the conversion from the native units (typically
+        # radians) to degrees for the first observation.  This allows us
+        # to treate ra/dec as numpy arrays rather than Angles, which would
+        # be approximately 600x slower.
+
+        conv = refCat[0]['coord_ra'].asDegrees() / float(refCat[0]['coord_ra'])
+        fgcmRefCat['ra'] = refCat['coord_ra'][selected] * conv
+        fgcmRefCat['dec'] = refCat['coord_dec'][selected] * conv
+
+        # Default (unset) values are 99.0
+        fgcmRefCat['refMag'][:, :] = 99.0
+        fgcmRefCat['refMagErr'][:, :] = 99.0
+
+        if self.config.applyColorTerms:
+            try:
+                refCatName = self.refObjLoader.ref_dataset_name
+            except AttributeError:
+                # NOTE: we need this try:except: block in place until we've
+                # completely removed a.net support
+                raise RuntimeError("Cannot perform colorterm corrections with a.net refcats.")
+
+            for i, (filterName, fluxField) in enumerate(zip(self._fluxFilters, self._fluxFields)):
+                if fluxField is None:
+                    continue
+
+                self.log.debug("Applying color terms for filtername=%r" % (filterName))
+
+                colorterm = self.config.colorterms.getColorterm(
+                    filterName=filterName, photoCatName=refCatName, doRaise=True)
+
+                refMag, refMagErr = colorterm.getCorrectedMagnitudes(refCat, filterName)
+
+                good, = np.where((np.nan_to_num(refMag[selected]) < 90.0) &
+                                 (np.nan_to_num(refMagErr[selected]) < 90.0) &
+                                 (np.nan_to_num(refMagErr[selected]) > 0.0))
+
+                fgcmRefCat['refMag'][good, i] = refMag[selected][good]
+                fgcmRefCat['refMagErr'][good, i] = refMagErr[selected][good]
+
+        else:
+            # No colorterms
+
+            # TODO: need to use Jy here until RFC-549 is completed and refcats return nanojansky
+
+            for i, (filterName, fluxField) in enumerate(zip(self._fluxFilters, self._fluxFields)):
+                good, = np.where((np.nan_to_num(refCat[fluxField][selected]) > 0.0) &
```

Comment:
Is this essentially saying: "where (not nan) and  (> 0)"?

Reply:
Yup.  I've added a comment that `nan_to_num` converts `nan`s to zeros, which is conveniently what we want in this selection.